_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,428 @@
+
+It has three different actions:
+
+| Action | Description |
+| :-----------: | ------------- |
+| `generate`  | Takes a set of topics and brokers and generates a reassignment JSON file. This actions is optional if we already have `reassignment.json` file with us. You can use this actions using `--generate`|
+| `execute` | Takes a reassignment JSON file and applies it to the partitions and brokers in the cluster. The `reassignment.json` file can be either the one proposed by the `generate` action or written by the user itself. `--execute`  is used to carry out this action |
+| `verify`  | Using the same reassignment JSON file as the `--execute` step, `--verify` checks whether all the partitions in the file have been moved to their intended brokers. If the reassignment is complete, `--verify` also removes any replication quotas (`--throttle`) that are in effect. Unless removed, throttles will continue to affect the cluster even after the reassignment has finished. This action can be executed using `--verify` |
+
+## Example time
+
+Let us understand the working of this tool with a concrete example.
+Suppose we have 5 Kafka Brokers and after looking at the partition details we realize that the brokers are not too busy, so we can scale them down to 3.
+Through this example we will take a look at how the three actions of the Kafka reassignment partition tool(`--generate`, `--execute` and `--verify`) works.
+We will generate the JSON data that will be used in the `reassignment.json` file.
+We will then assign the partitions to the remaining broker using the `reassignment.json` file.
+
+Before proceeding towards the steps. Let's discuss one more curious question. Can you scale down any pod you want through this process?
+So the answer to this question is no. Wondering why?
+It is due to the fact Strimzi uses StatefulSets to manage broker pods. So you cannot remove any pod from the cluster.
+You can only remove one or more of the highest numbered pods from the cluster. For example, in a cluster of 5 brokers the pods are named `<CLUSTER-NAME>-kafka-0` up to `<CLUSTER-NAME>-kafka-4`.
+If you decide to scale down by two brokers, then `<CLUSTER-NAME>-kafka-4` and `<CLUSTER-NAME>-kafka-3` will be removed.
+
+ The next section will help you set up the environment for executing the above example ie setting up your kafka cluster, kafka topics and also configuring the Kafka user with required ACL's.
+
+Note: In case you already have the environment set up, you can skip the next section.
+
+### Setting up the environment
+
+To get the kafka cluster up and running , we will first have to install the Strimzi Cluster Operator and then deploy the Kafka resource.
+You can install the Cluster Operator with any installation method you prefer.
+The Kafka cluster is then deployed with the plain listener enabled on port 9092.
+
+Example Kafka configuration with plain listener.
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: <CLUSTER-NAME>
```

Comment:
Can we put a default name? That would allow users to just copy and paste all these examples